### PR TITLE
Add evnt listener hook in base-shell, for reused. Upd Locale. integrating load provider with demo in base-shell

### PR DESCRIPTION
**Pull request by UseMuse**

### CODE REVIEW

**Comment by UseMuse** [comment]
Rename

File: packages/base-shell/src/providers/Load/Provider.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    LOADING: useCallback((displayText) => {
+      _dispatch({ type: 'LOADING', payload: { displayText } });
+    }, [_dispatch]),
+    LOADED:  useCallback((payload) => {
+      _dispatch({ type: 'LOADED', payload: payload });
+    }, [_dispatch]),
+    LOAD_CANCELED:  useCallback((payload) => {
+      _dispatch({ type: 'LOAD_CANCELED', payload: payload });
+    }, [_dispatch]),
+    LOADED_ERROR:  useCallback((payload) => {
```
Comment (UseMuse) [inline packages/base-shell/src/providers/Load/Provider.js:49]: loading with an error, 2 seconds to show a red preloader, to display an error loading

File: packages/base-shell/src/providers/Load/Provider.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      _dispatch({ type: 'LOADING_WITH_ERROR', payload: payload });
```
Comment (UseMuse) [inline packages/base-shell/src/providers/Load/Provider.js:50]: loading with an error, 2 seconds to show a red preloader, to display an error loading

File: packages/base-shell/src/providers/Load/Provider.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      sleep().then(() => _dispatch({ type: 'LOADED_ERROR', payload: payload }))
```
Comment (UseMuse) [inline packages/base-shell/src/providers/Load/Provider.js:51]: loading with an error, 2 seconds to show a red preloader, to display an error loading

**Comment by TarikHuber** [comment]
Could you pls add the option for loadText to be changed. We can't use always english. You can leave the english text as deafult but tehre should be an option to use custom labels for different locales.

**Comment by UseMuse** [comment]
Ok

**Comment by UseMuse** [comment]
Either that, or you need to raise above LocalProvider, above LoadProvider

**Comment by TarikHuber** [comment]
That solution with intl is to complicated. Just make it use label strings. The default intl change will rerender the whole app and change thoes if needed.

**Comment by UseMuse** [comment]
@TarikHuber I'm very busy so far, I would be grateful if you try to do this

**Pull request closed without merging**